packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -346,22 +346,6 @@ export class ProjectActions {
       throw Error(`Cannot create spec without currentProject.`)
     }
 
-    const parsed = path.parse(codeGenCandidate)
-
-    const possibleExtensions = ['.cy', '.spec', '.test', '-spec', '-test', '_spec']
-
-    const getFileExtension = () => {
-      if (erroredCodegenCandidate) {
-        return ''
-      }
-
-      return (
-        possibleExtensions.find((ext) => {
-          return codeGenCandidate.endsWith(ext + parsed.ext)
-        }) || parsed.ext
```

Comment:
This `|| parsed.ext` fallback was the root of the issue and was replaced with `|| ''` in the moved logic here: https://github.com/cypress-io/cypress/pull/21593/files#diff-1dbda8880bf7cf51476b46c4e8344284db3c540b89672fe2dc63b3d61ae8228fR45